recent string loading changes break big endian with mojibake

## Steps to Reproduce

1. build mono


### Current Behavior

[mono hid the facts](https://en.wikipedia.org/wiki/Bush_hid_the_facts)

```
13:46:29 MONO_PATH=".//class/lib/monolite-linux/2EB20924-0AB7-455D-BCFB-0E4A886B6EF8:.//class/lib/monolite-linux/2EB20924-0AB7-455D-BCFB-0E4A886B6EF8/Facades:$MONO_PATH" /home/jenkins/workspace/test-mono-mainline-community/label/centos-s390x/runtime/mono-wrapper  .//class/lib/monolite-linux/2EB20924-0AB7-455D-BCFB-0E4A886B6EF8/mcs.exe /warn:0 /noconfig /langversion:latest /r:System.dll /r:mscorlib.dll /out:build/deps/basic-profile-check.exe.tmp build/common/basic-profile-check.cs
13:46:29 Corlib not in sync with this runtime: The runtime did not find the mscorlib.dll it expected. Expected interface version 2EB20924-0AB7-455D-BCFB-0E4A886B6EF8 but found ㈀䔀䈀㈀　㤀㈀㐀ⴀ　䄀䈀㜀ⴀ㐀㔀㔀䐀ⴀ䈀䌀䘀䈀ⴀ　䔀㐀䄀㠀㠀㘀䈀㘀䔀䘀㠀. Check that your runtime and class libraries are matching.
13:46:29 Loaded from: /home/jenkins/workspace/test-mono-mainline-community/label/centos-s390x/mcs/class/lib/monolite-linux/2EB20924-0AB7-455D-BCFB-0E4A886B6EF8/mscorlib.dll
13:46:29 Download a newer corlib or a newer runtime at http://www.mono-project.com/download.
13:46:29 
make[8]: *** [build/deps/basic-profile-check.exe] Error 1
```

### Expected Behavior

build proceeds as normal

## On which platforms did you notice this

[ ] macOS
[X] Linux
[ ] Windows
[X] AIX

**Version Used**:

```
Mono JIT compiler version 5.19.0 (master/d36902e Mon Aug 13 11:43:24  2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:
        SIGSEGV:       normal
        Notification:  Thread + polling
        Architecture:  ppc
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        Suspend:       preemptive
        GC:            sgen
```

--- a/mono/metadata/object.c
+++ b/mono/metadata/object.c
@@ -7396,7 +7396,19 @@ mono_string_from_blob (const char *str, MonoError *error)
 {
 	gsize len = mono_metadata_decode_blob_size (str, &str) >> 1;
 
+#if G_BYTE_ORDER != G_LITTLE_ENDIAN
+	gunichar2 *src = (gunichar2*)str;
+	gunichar2 *copy = g_new (gunichar2, len);
+	int i;
+	for (i = 0; i < len; ++i)
+		copy [i] = GUINT16_FROM_LE (src [i]);
+
+	char *res = mono_utf16_to_utf8 (copy, len, error);
+	g_free (copy);
+	return res;
+#else
 	return mono_utf16_to_utf8 ((const gunichar2*)str, len, error);
+#endif
 }
 /**
  * mono_ldstr_metadata_sig

